optimum/utils/preprocessing/image_classification.py

```diff
@@ -0,0 +1,102 @@
+from functools import partial
+from typing import Dict, List
+
+import torch
+from datasets import Dataset, Metric, load_dataset
+from torchvision.transforms import CenterCrop, Compose, Normalize, Resize, ToTensor
+from transformers import FeatureExtractionMixin, ImageClassificationPipeline
+
+from .base import DatasetProcessing
+
+
+class ImageClassificationProcessing(DatasetProcessing):
+    def __init__(self, **kwargs):
+        if "secondary" in kwargs["data_keys"]:
+            raise ValueError("Only one data column is supported for token-classification.")
+        else:
+            kwargs["data_keys"]["secondary"] = None
+
+        super().__init__(**kwargs)
+
+        if not isinstance(self.preprocessor, FeatureExtractionMixin):
+            raise ValueError(
+                f"Preprocessor is expected to be a feature extractor, provided {type(self.preprocessor)}."
+            )
+
+    def load_datasets(self):
+        # Downloading and loading a dataset from the hub.
+        raw_datasets = load_dataset(path=self.dataset_path, name=self.dataset_name)
+
+        max_eval_samples = 100  # TODO remove this
```

Comment:
To remove I guess

Reply:
Thank you! It is removed in the next PR where `max_eval_samples` is provided as an argument.